clang/test/CodeGen/cheri/riscv/non-cheri-march.c

```diff
@@ -0,0 +1,11 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
+/// Using __capability without the appropriate -march string previously
+/// crashed clang during codegen instead of emitting an error
+// RUN: %clang_cc1 -triple riscv64 -o - -emit-llvm %s -verify=no-cheri
+// RUN: %clang_cc1 -triple riscv64 -target-feature +xcheri -o - -emit-llvm %s | FileCheck %s
+
+// CHECK-LABEL: @test(
+// CHECK-NEXT:  entry:
+// CHECK-NEXT:    ret i8 addrspace(200)* null
+//
+void *__capability test(void) { return 0; } // no-cheri-error{{'__capability' attribute is not supported on targets missing CHERI; specify an appropriate -march= or -mcpu=}}
```

Comment:
`void * __capability p;` as a global is probably the simplest you can get?

Reply:
That is what creduce ended up giving me, I can't remember why I used a function instead.

Reply:
Probably it just happened to end up with a function rather than a global as they're much more prevalent and so statistically you're more likely to pick that as the reduced code

Reply:
Oh I see what you mean, hmm